core/src/com/unciv/models/ruleset/tile/TileResource.kt

```diff
@@ -141,6 +157,13 @@ class TileResource : RulesetStatsObject() {
         }
     }
 
+    fun matchesFilter(filter: String) = when (filter) {
+        name -> true
+        "any" -> true
+        resourceType.name -> true
+        else -> improvementStats?.any { filter == it.key.name } == true
```

Comment:
I'm not sure I agree with this one, and it's missing the most important filter - uniques contains(filter)